### ISSUE TITLE
Size of CoqIDE panels is uneven

### ISSUE THREAD

**Issue by robbertkrebbers** [coq/coq#10956]
When starting CoqIDE, creating a new file, or opening a file, the size of the panels is uneven, see below for a screenshot:

![coqide](https://robbertkrebbers.nl/misc/coqide.png)

I would expect that the panels are 50% in horizontal and vertical size.

Note that even if I resize the panels to something reasonable, they are reset when creating a new file or opening a file.

#### Coq Version

8.10 on Debian unstable.

**Comment by silene** [comment]
Dimensions of the panels are per file. So, by reset, do you mean that the newly opened files have uneven dimensions? Or do you mean that even the already opened files get resized to uneven dimensions?

**Comment by robbertkrebbers** [comment]
The former, the newly opened files have uneven dimensions. The dimensions of the already opened files stay the same.

**Comment by herbelin** [comment]
@silene: Maybe a stupid question, do you know a way to force a dummy event from gtk itself (as a way to force an iteration of the `Events`, `Update`, `Layout`, `Draw` loop that you were mentioning in #10748)? That could then maybe provide a more robust way to positioning the handle in the middle?

**Comment by silene** [comment]
There are the `*_put_event` functions, but as far as I know, they are not exported by lablgtk.

Also, if the `coqiderc` file of the user contains
```
window_height = "600" 
window_width = "800" 
```
I am not sure what we are supposed to do. Should we just disregard the user settings?

**Comment by silene** [comment]
Just to be clearer with my answer, there are two issues there:
1. The window is created accordingly to the user settings, but if the window manager does not respect the user settings, we have no way of guessing what will be the size of the window eventually.
2. New session panes should not look at the user settings but at the current window size, as the user might have resized the window since then.

**Comment by cogumbreiro** [comment]
> Just to be clearer with my answer, there are two issues there:
> 
>     1. The window is created accordingly to the user settings, but if the window manager does not respect the user settings, we have no way of guessing what will be the size of the window eventually.
> 
>     2. New session panes should not look at the user settings but at the current window size, as the user might have resized the window since then.

I apologize, but I don't understand the answer 2. I can replicate the behaviour posted by OP.

**Could it be possible that CoqIDE is storing the _initial_ window dimensions (versus the current window dimensions)?**

When I open CoqIDE the windows starts this big:

![Screenshot from 2019-11-06 14-11-18](https://user-images.githubusercontent.com/2490181/68329722-de717900-009f-11ea-8018-77e75ebeea78.png)

Then, I maximize the window and get a good ratio:

![Screenshot from 2019-11-06 14-11-26](https://user-images.githubusercontent.com/2490181/68329687-ce599980-009f-11ea-8902-4bf01dad1963.png)
 
However, once I create a new tab the behaviour posted by OP appears:

![Screenshot from 2019-11-06 14-11-35](https://user-images.githubusercontent.com/2490181/68329781-03fe8280-00a0-11ea-86e7-4fe53f78afd8.png)

It looks like the editor pane has the same dimensions as in the first screenshot.

**Comment by silene** [comment]
I guess I was still not clear enough. So, let me try again. Coqide 8.10 always trusts the content of the `coqiderc` user configuration file to decide the dimensions of the panes. This was a dumb idea, sorry.

**Comment by cogumbreiro** [comment]
Thanks for the clarification @silene! I really appreciate it.

Let me see if I understood. The default size of the window is 800x600 and that is written to `~/.config/coq/coqiderc`. The dimensions stored in `coqiderc` is used when creating a new tab, yet for some reason the algorithm only only cares about the width and ignores the height.

**Comment by silene** [comment]
As far as I can tell, all the screenshots in this issue are consistent with a height of 600. What makes you think that the height is ignored?

**Comment by cogumbreiro** [comment]
I meant that the height is ignored when the window is maximized. If I unmaximize the window, then the window goes back to the 800x600 which I can confirm is also in `coqiderc`. The width of the editor pane is also reduced as you can see in the following screenshot.
![Screenshot from 2019-11-06 17-55-11](https://user-images.githubusercontent.com/2490181/68345163-b8a79c80-00be-11ea-98e1-5607a64395fd.png)

**Comment by silene** [comment]
Oh! This one is easy and no specific algorithm is involved. This is just a matter of choosing which behavior we want when the window's height is increased:

1. the upper pane's height is increased and the lower pane's height stays the same,
2. the lower pane's height is increased and the upper pane's height stays the same,
3. both upper and lower panes' height are increased.

The default behavior of GTK is 2, but we can ask for any of the two other behaviors. (If it was my choice alone, I would go for for behavior 1.)

Conversely, when the height is decreased, there are also three available behaviors.
